ui/src/ducks/app/monitoring.js

```diff
@@ -336,18 +337,18 @@ export function* fetchVolumeStats() {
   const startingTimeISO = new Date(startingTimestamp * 1000).toISOString();
   const volumeUsedQuery = 'kubelet_volume_stats_used_bytes';
 
-  // the queries for Throughput/IOPS
+  // the queries for `Throughput` and `IOPS`
   // rate calculates the per-second average rate of increase of the time series in the range vector.
   // group the result of the query by instance and device (remove the filter {job="node-exporter"})
   const volumeThroughputReadQuery = `sum(irate(node_disk_read_bytes_total[1m])) by (instance, device)`;
   const volumeThroughputWriteQuery = `sum(irate(node_disk_written_bytes_total[1m])) by (instance, device)`;
   const volumeIOPSReadQuery = `sum(irate(node_disk_reads_completed_total[5m])) by (instance, device)`;
   const volumeIOPSWriteQuery = `sum(irate(node_disk_writes_completed_total[5m])) by (instance, device)`;
 
-  // the query for latency
+  // the query for `latency`
   // Disk latency is the time that it takes to complete a single I/O operation on a block device
-  // rate(node_disk_read_time_seconds_total[5m]) / rate(node_disk_reads_completed_total[5m]
-  const volumeLatencyQuery = `sum(irate(node_disk_io_time_seconds_total[1m])) by (instance, device)`;
+  const volumeLatencyWriteQuery = `sum(irate(node_disk_write_time_seconds_total[5m]) / irate(node_disk_writes_completed_total[5m])) by (instance, device)`;
```

Comment:
Are you certain we want to use `irate` here? I don't have a clear answer for you, so I suggest we keep it like this for now and open a ticket to check what this query (and others) mean exactly.

FYI, computing rates and means isn't an easy task (at all) and can be super counter-intuitive. Here are some valuable articles to get some insights on the topics:
- On `rate` vs. `irate`: https://utcc.utoronto.ca/~cks/space/blog/sysadmin/PrometheusRateVsIrate
- On the three Pythagorean means: https://towardsdatascience.com/on-average-youre-using-the-wrong-average-geometric-harmonic-means-in-data-analysis-2a703e21ea0

Reply:
Hello @gdemonet ,
I also have an article to share [Irate graphs are better graphs](https://www.robustperception.io/irate-graphs-are-better-graphs)

> The rate function takes a time series over a time range, and based on the first and last data points within that range (allowing for counter resets) calculates a per-second rate. 

> By contrast irate is an instant rate. It only looks at the last two points within the range passed to it and calculates a per-second rate.

So by using irate we can see much briefer spikes and finer detail.